framework/include/materials/Material.h

```diff
@@ -217,6 +217,8 @@ class Material : public MaterialBase, public Coupleable, public MaterialProperty
   /// resolve all optional properties
   virtual void resolveOptionalProperties() override;
 
+  virtual bool shouldWarn() const override { return !_bnd || boundaryRestricted(); }
```

Comment:
The concern I have with this is what if a developer warns for some condition that is only triggered in a face or boundary evaluation?

Reply:
I think that would be a rare case compared with this. A developer can override this function again in his material. Note that the logic here is to warn on element and boundary materials except face and neighbor.